src/ngx_http_modsecurity_module.c

```diff
@@ -766,5 +791,50 @@ ngx_http_modsecurity_cleanup_rules(void *data)
     ngx_http_modsecurity_pcre_malloc_done(old_pool);
 }
 
+static int
+ngx_http_modsecurity_set_up_log_reload(ngx_conf_t *cf, ngx_http_modsecurity_conf_t *mcf)
+{
+    ngx_http_modsecurity_main_conf_t  *mmcf;
+    ngx_list_t                        *list;
+    Rules                         **item;
+    
+    mmcf = ngx_http_conf_get_module_main_conf(cf, ngx_http_modsecurity_module);
+    list = mmcf->audit_log_reload->data;
+
+    // Each rules set may have an audit log. We need to remember each rules set
+    // so we can ask for its audit log to be reopened.
+    item = ngx_list_push(list);
+    if (item == NULL) {
+        dd("failed to append a rules set for log reload");
+        return -1;
+    }
+    *item = mcf->rules_set;
+
+    return 0;
+}
+
+static void
+ngx_http_modsecurity_log_reload(ngx_open_file_t *file, ngx_log_t *log)
+{
+    ngx_list_t       *list;
+    ngx_list_part_t  *part;
+    ngx_uint_t        i;
+    Rules        **rules_sets;
+    const char       *error = NULL;
+
+    ngx_log_debug0(NGX_LOG_DEBUG_EVENT, log, 0, "modsecurity audit log buffer flush");
+
+    list = file->data;
+
+    for (part = &list->part; part != NULL; part = part->next) {
+        rules_sets = part->elts;
+        for (i = 0; i < part->nelts; i++) {
+            if (msc_rules_reopen_audit_log(rules_sets[i], &error) < 0) {
```

Comment:
The other call to this function results in a catastrophic failure, but this one will only log the error and keep going. Not sure if this is a problem or not, just pointing it out.

Reply:
I think this is desired behavior.

In the first case, you have an opportunity to fail early, including failing to reload a broken config. nginx should continue running with the old config in that case. I'm planning to test this belief.

In the second case, you've already successfully loaded a config and are simply trying to reopen log files. If that fails, it seems reasonable and desired for nginx to continue running.

Reply:
> In the first case, you have an opportunity to fail early, including failing to reload a broken config. nginx should continue running with the old config in that case.

Confirmed. It's probably documented somewhere, but I just tested.